service/storage_service.cc

```diff
@@ -1419,29 +1431,47 @@ class topology_coordinator {
             }
         });
 
-        if (needs_barrier) {
-            guard = co_await global_tablet_token_metadata_barrier(std::move(guard));
-        }
-
         // In order to keep the cluster saturated, ask the load balancer for more transitions.
         // Unless there is a pending topology change operation.
+        auto ts = guard.write_timestamp();
```

Comment:
With `guard` basically being optional object now, even innocuous `write_timestamp()` call may be unsafe :(